### PR TITLE
Assignment 4 aajodar

### PR DESCRIPTION
**Pull request by aajodar**

### CODE REVIEW

File: Assignment4/aajodar-22A243/Task07.py [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+
+# In[9]:
+
+
+# SPARQL
+from rdflib.plugins.sparql import prepareQuery
+
+q1 = prepareQuery('''
+  SELECT ?s WHERE { 
+    ?s RDFS:subClassOf ns:LivingThing .
```
Comment (dgarijo) [inline Assignment4/aajodar-22A243/Task07.py:47]: what about the subclasses of subclasses of living thing? And so on...

File: Assignment4/aajodar-22A243/Task07.py [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+
+# In[8]:
+
+
+# RDFLib
+ns = Namespace("http://somewhere#")    
+
+# Visualize the results
+for s,p,o in g.triples((None, RDFS.subClassOf, ns.LivingThing)):
+    print(s)
```
Comment (dgarijo) [inline Assignment4/aajodar-22A243/Task07.py:36]: Nice but make it a recursive function to get all the tree of subclasses

File: Assignment4/aajodar-22A243/Task07.py [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+ns = Namespace("http://somewhere#")    
+
+# Individuals of "Person"
+for s,p,o in g.triples((None, RDF.type, ns.Person)):
+    print(s)
+
+# Individuals of the subclases of "Person"
+for subclass,_,_ in g.triples((None, RDFS.subClassOf, ns.Person)):
+    for ind,_,_ in g.triples((None, RDF.type, subclass)):
+        print(ind)
```
Comment (dgarijo) [inline Assignment4/aajodar-22A243/Task07.py:74]: This only gets you the subclasses of person, not the subclasses of subclasses like the query below.
The one in sparql is correct, this one is missing small additions

File: Assignment4/aajodar-22A243/Task07.py [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+# In[37]:
+
+
+# SPARQL
+q3 = prepareQuery('''
+  SELECT DISTINCT ?ind ?prop WHERE { 
+    {
+    ?ind RDF:type ns:Person . 
+    } UNION {
+    ?s RDFS:subClassOf* ns:Person .
```
Comment (dgarijo) [inline Assignment4/aajodar-22A243/Task07.py:139]: the exercise explicitly states that it's not needed to get all the subclass hierarchy...

**Review by dgarijo — CHANGES_REQUESTED** [review]
Very small change requests, it's almost perfect. Good job

**Comment by SerPablo** [comment]
Some minor errors. No more PRs allowed

**Pull request merged**